src/vs/editor/contrib/gotoSymbol/browser/goToCommands.ts

```diff
@@ -229,10 +229,14 @@ export abstract class SymbolNavigationAction extends EditorAction {
 		return targetEditor;
 	}
 
-	private _openInPeek(target: ICodeEditor, model: ReferencesModel) {
+	private _openInPeek(target: ICodeEditor, model: ReferencesModel, range?: Range) {
 		const controller = ReferencesController.get(target);
 		if (controller && target.hasModel()) {
-			controller.toggleWidget(target.getSelection(), createCancelablePromise(_ => Promise.resolve(model)), this.configuration.openInPeek);
+			if (range) {
+				controller.toggleWidget(range, createCancelablePromise(_ => Promise.resolve(model)), this.configuration.openInPeek);
+			} else {
+				controller.toggleWidget(target.getSelection(), createCancelablePromise(_ => Promise.resolve(model)), this.configuration.openInPeek);
+			}
```

Comment:
Nicer, shorter would be `range = range ?? target.getSelection()` and later always use the range. That would be less similar/duplicated code